### ISSUE TITLE
clippy::len_without_is_empty bug when implementing on a type alias

### ISSUE THREAD

**Issue by bobogei81123** [rust-lang/rust-clippy#11165]
### Summary

When a type alias is created, say `type B = A;` and both `len` and `is_empty` are implemented on `B`, clippy still complain that type `B` has a public `len` method, but no `is_empty` method  .

### Reproducer

I tried this code:

```rust
pub struct A;
pub type B = A;

impl B {
    /// Common method
    pub fn len(&self) -> usize {
        todo!()
    }

    pub fn is_empty(&self) -> bool {
        todo!()
    }
}
```

I expected to see this happen: No warnings, specifically, `clippy::len_without_is_empty`, is emitted.

Instead, this happened: Clippy returns the warning  `clippy::len_without_is_empty`.

Clippy version:

```shell
$ cargo clippy --version               
clippy 0.1.72 (ad96323 2023-07-14)
```

Result of running `cargo clippy`:

```shell
$ cargo clippy          
warning: type `B` has a public `len` method, but no `is_empty` method
 --> src/lib.rs:6:5
  |
6 |     pub fn len(&self) -> usize {
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#len_without_is_empty
  = note: `#[warn(clippy::len_without_is_empty)]` on by default

warning: `rust-doc-repro` (lib) generated 1 warning
    Finished dev [unoptimized + debuginfo] target(s) in 0.02s
  ```

### Version

```text
rustc 1.73.0-nightly (ad963232d 2023-07-14)
binary: rustc
commit-hash: ad963232d9b987d66a6f8e6ec4141f672b8b9900
commit-date: 2023-07-14
host: x86_64-unknown-linux-gnu
release: 1.73.0-nightly
LLVM version: 16.0.5
```
